sim/priest/power_infusion.go

```diff
@@ -26,7 +28,14 @@ func (priest *Priest) registerPowerInfusionCD() {
 		UsesGCD:    true,
 		Type:       core.CooldownTypeMana,
 		CanActivate: func(sim *core.Simulation, character *core.Character) bool {
-			// Does there need to be logic around this?
+			if powerInfusionTarget == nil {
+				return false
+			}
+			// How can we determine the target will be able to continue casting 
+			// 	for the next 15s at 20% reduced mana cost? Arbitrary value until then.
+			if powerInfusionTarget.CurrentMana() < 3000 {
```

Comment:
Ah, this is a check that would make more sense in ShouldActivate. CanActivate should check whether the casting priest has enough mana to cast the PI spell.